lib/db/instance.go

```diff
@@ -34,7 +34,7 @@ func (db *instance) updateFiles(folder, device []byte, fs []protocol.FileInfo, m
 	t := db.newReadWriteTransaction()
 	defer t.close()
 
-	var dk, gk []byte
+	var dk, gk, keyBuf []byte
```

Comment:
The same thing could be achieved with just two buffers and the naming isn't as succinct as possible (keyBuf vs just buf), but this choice will become more clear in the followup PR (with blockBuf). Having three buffers also is a bit safer, as then `dk` really only is used as device-file-key, not for other purposes too.

Reply:
I think having several is fine. Less risk that we reuse it half way in some function call and then expect it to be unchanged on the next line.